### PR TITLE
PERF: allow to skip validation/sanitization in DataFrame._from_arrays

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
For cases where you know to have valid data (eg you just created them yourself, or they are already validated), it can be useful to skip the validation checks when creating a DataFrame from arrays.

Use case is for example https://github.com/pandas-dev/pandas/pull/32825

From investigating https://github.com/pandas-dev/pandas/issues/32196#issuecomment-600824238

@rth this gives another 20% improvement on the dataframe creation part. Together with https://github.com/pandas-dev/pandas/pull/32856, it gives a bit more than a 2x improvement on the dataframe creation part (once the sparse arrays are created)

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
```
In [1]: arrays = [pd.arrays.SparseArray(np.random.randint(0, 2, 1000), dtype="float64") for _ in range(10000)] 
   ...: index = pd.Index(range(len(arrays[0])))   
   ...: columns = pd.Index(range(len(arrays)))

In [2]: %timeit pd.DataFrame._from_arrays(arrays, index=index, columns=columns)   
119 ms ± 3.52 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [3]: %timeit pd.DataFrame._from_arrays(arrays, index=index, columns=columns, verify_integrity=False)    
98.1 ms ± 713 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

**Comment by rth** [comment]
Very nice, thanks @jorisvandenbossche ! It's good that is applies to extension arrays in general not just sparse frames. Though I guess the use case of a very large number of columns (>10k) is less common outside of sparse.

**Comment by rth** [comment]
Actually my comment was more about https://github.com/pandas-dev/pandas/pull/32856, got confused in your multiple performance improvement PRs :) This is nice too!

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1889,8 +1889,40 @@ def to_records(
...
+            Each array in the list corresponds to one column, in order.
+        columns : list-like, Index
+            The column names for the resulting DataFrame.
+        index : list-like, Index
+            The rows labels for the resulting DataFrame.
+        dtype : dtype, optional
+            Optional dtype to enforce for all arrays.
+        verify_integrity : bool, default True
+            Validate and homogenize all input. If set to False, it is assumed
+            that all elements of `arrays` are actual arrays to be stored in
```
Comment (rth) [inline pandas/core/frame.py]: By "actual arrays" do you mean numpy ndarray or pandas arrays? Might be worth specifying.

**Comment by jorisvandenbossche** [comment]
I think your comment applies to both ;)

File: pandas/core/frame.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/frame.py]: One of both. Basically the array as how it is stored in a block. Will mention it needs to be one of both.

**Comment by simonjayhawkins** [comment]
Nice docstring. 😄

**Review by TomAugspurger — APPROVED** [review]
What order do we want to do this and https://github.com/pandas-dev/pandas/pull/32825 in?

Seems like we merge this first and then update https://github.com/pandas-dev/pandas/pull/32825 to pass `verify_integrity=False`?

**Comment by jorisvandenbossche** [comment]
Doesn't matter too much, can be added either here or there. 
And I would also like to add `verify_integrity` to the benchmark that I am adding in https://github.com/pandas-dev/pandas/pull/32856

**Comment by TomAugspurger** [comment]
SGTM. I think this can go in since CI is passing :)

**Pull request merged**